pkg/syncer/mutators/deployment.go

```diff
@@ -0,0 +1,247 @@
+package mutators
+
+import (
+	"net"
+	"net/url"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	utilspointer "k8s.io/utils/pointer"
+)
+
+type DeploymentMutator struct {
+	fromConfig *rest.Config
+	kcpHost    string
+}
+
+func (dt *DeploymentMutator) getGVR() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    "apps",
+		Version:  "v1",
+		Resource: "deployments",
+	}
+}
+
+func (dt *DeploymentMutator) Register(mutators map[schema.GroupVersionResource]Mutator) {
+	if _, ok := mutators[dt.getGVR()]; !ok {
+		mutators[dt.getGVR()] = dt
+	}
+}
+
+func NewDeploymentMutator(fromConfig *rest.Config, kcpHost string) *DeploymentMutator {
+	return &DeploymentMutator{
+		fromConfig: fromConfig,
+		kcpHost:    kcpHost,
+	}
+}
+
+// ApplySpec makes modifications to the Spec of the deployment object.
+func (dt *DeploymentMutator) Mutate(downstreamObj *unstructured.Unstructured) error {
+	var deployment appsv1.Deployment
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(
+		downstreamObj.UnstructuredContent(),
+		&deployment)
+	if err != nil {
+		return err
+	}
+
+	// If the deployment has not serviceaccount defined or it is "default", that means that it is using the default one
+	// and we need will need to override it with our own, "kcp-default"
+	//
+	// If the deployment has a serviceaccount defined, that means we need to synchronize the created service account
+	// down to the workloadcluster, so we don't modify it, and expect the scheduler to do the job.
+	//
+	if deployment.Spec.Template.Spec.ServiceAccountName == "" || deployment.Spec.Template.Spec.ServiceAccountName == "default" {
```

Comment:
(No action required): So many dots! Maybe alias the template spec to simplify things?

```golang
templateSpec := deployment.Spec.Template.Spec
```

Reply:
done, thanks.